OpenSim/Common/Component.h

```diff
@@ -1577,10 +1595,82 @@ template <class T> friend class ComponentMeasure;
         of StateVariables by name. 
         
         NOTE: If the component name or the state variable name is ambiguous, the 
-         first instance found is returned. To disambiguate use the full name provided
+         an exception is thrown. To disambiguate use the full name provided
```

Comment:
Yes, there is a bug that it should verify against the the full path name instead of just assuming it is the top of a relative path in the return case, otherwise it should have added itself to the `foundCs` list. Thanks!

As background 1618 was added to handle the corner case where you are calling findComponent() on the component you actually wanted to find. It almost never happens (why would you call find for a Component if you already had a handle to it to ask it to find) but I thought it would be weird for it to say it could not find itself. It's not there for performance.